### PR TITLE
keadm Support centos

### PR DESCRIPTION
**Pull request by FengyunPan2**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

> /kind feature

**What this PR does / why we need it**:
 1. fill the functions of centosinstaller.
 2. add check before install service or download file
 3. split some commands to fix the issue of 'chmod +x ...' which can't
    work.

**Special notes for your reviewer**:

/assign @fisherxu 

**Does this PR introduce a user-facing change?**:

```release-note
keadm Support centos.
```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/util/centosinstaller.go [hunk 1]
```diff
@@ -40,57 +42,54 @@ func (c *CentOS) SetKubeEdgeVersion(version string) {
...
+
+	serverVersion, err := discoveryClient.ServerVersion()
+	if err != nil {
+		return fmt.Errorf("Failed to get the version of K8s master, please check whether K8s was successfully installed, err: %v", err)
+	}
+
+	k8sMinorVersion, _ := strconv.Atoi(serverVersion.Minor)
+	if k8sMinorVersion < types.DefaultK8SMinimumVersion {
+		return fmt.Errorf("Your minor version of K8s is lower than %d, please reinstall newer version", types.DefaultK8SMinimumVersion)
+	}
```
[the remark this answers is not in the document]
Comment (FengyunPan2) [reply keadm/cmd/keadm/app/cmd/util/centosinstaller.go]: Just a little.

File: keadm/cmd/keadm/app/cmd/util/centosinstaller.go [hunk 2]
```diff
@@ -99,22 +98,180 @@ func (c *CentOS) IsK8SComponentInstalled(component, defVersion string) error {
...
+		}
+		fmt.Println(stdout)
+	}
+	if componentType == types.EdgeCore {
+		stdout, err := runCommandWithStdout(untarFileAndMoveEdgeCore)
+		if err != nil {
+			return err
+		}
+		fmt.Println(stdout)
+	}
```
[the remark this answers is not in the document]
Comment (FengyunPan2) [reply keadm/cmd/keadm/app/cmd/util/centosinstaller.go]: Done.

File: keadm/cmd/keadm/app/cmd/util/centosinstaller.go [hunk 3]
```diff
@@ -40,81 +37,72 @@ func (c *CentOS) SetKubeEdgeVersion(version string) {
...
+	if err != nil {
+		return err
+	}
+
+	// TODO: may support more case
+	switch result {
+	case "x86_64":
+		arch = "amd64"
+	default:
+		arch = "unknown"
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/centosinstaller.go]: Could we return err here directly?

Comment (FengyunPan2) [reply keadm/cmd/keadm/app/cmd/util/centosinstaller.go]: Done.

File: keadm/cmd/keadm/app/cmd/util/k8sinstaller.go [hunk 4]
```diff
@@ -82,10 +82,17 @@ func installCRDs(kubeConfig, master string) error {
...
 		if err != nil {
+			if os.IsNotExist(err) {
+				//Download the tar from repo
+				dwnldURL := fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/%s", KubeEdgeCrdPath+"/devices", KubeEdgeCRDDownloadURL, crdFile)
+				_, err := runCommandWithShell(dwnldURL)
+				if err != nil {
+					return err
+				}
+			}
 			return err
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/k8sinstaller.go]: Need else here, or it will return the err after download the crd :)

Comment (FengyunPan2) [reply keadm/cmd/keadm/app/cmd/util/k8sinstaller.go]: My mistake, have tested it.

**Comment by kevin-wangzefeng** [comment]
@FengyunPan2 the PR needs rebase to fix CI failure.

**Pull request merged**